lib/count_cdn_logs.rb

```diff
@@ -0,0 +1,36 @@
+# coding: utf-8
+#
+# Count hits to each url in the log files
+#
+# Ensures that there are counts in the `<processed_dir>/raw_counts/` tree for
+# all log files which exist.
+#
+# If log files are removed, the counts are retained, so that we can archive the
+# logs but still do historical analysis of trends.
+
+require_relative 'config_logging'
+require_relative 'log_counter'
+
+class CountCdnLogs
+  def initialize(log_dir, processed_dir)
+    @log_dir = log_dir
+    @processed_dir = processed_dir
+    @counts_dir = "#{processed_dir}/raw_counts"
+  end
+
+  def update
+    files = completed_log_files
```

Comment:
I was actually wanting to avoid evaluating completed_log_files twice, in case the result was changed (eg, by an accidentally concurrently running program). But I failed to actually do that.

Instead, I'm changing this to memoize the result of completed_log_files, as well as calling it directly as suggested.